stdlib/map.ml

```diff
@@ -356,4 +360,42 @@ module Make(Ord: OrderedType) = struct
 
     let choose = min_binding
 
+    type 'a cursor_action =
+      | Act_yield of key * 'a
+      | Act_explore of 'a t
+
+    type 'a cursor = {
+      low: key option;
+      high: key option;
+      stack: 'a cursor_action list;
+    }
```

Comment:
I find the interval bounds overkill (and also it means you allocate twice as much on each `next` call). It's easy to implement a version that starts enumerating at a given key (or the smallest higher key) as a `cursor_start` variant, and easy to define a cuting-off variant of `cursor_next`.